### PR TITLE
Update and refactor code

### PR DESCRIPTION
**Pull request by theworkflow**
This PR does not change any functionality, but has breaking changes

- update codebase and tests to use es6 syntax
- deprecate node versions 0.10 and 4
- update dependencies and devDependencies
- update travis to test node versions 6, 7, and 8

### CODE REVIEW

**Comment by theworkflow** [comment]
@jackboberg ping

File: index.js [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+  if (isFunction(url)) [url, opts, done] = [null, {}, url]
+  if (isFunction(opts)) [opts, done] = [{}, opts]
 
   Amqp.connect(url, opts, function (err, conn) {
-    if (err) return done(err);
+    if (err) return done(err)
 
-    conn.createChannel(function (err, ch) {
-      if (err) return done(err);
+    const closeConnection = (err) => {
```
Comment (jackboberg) [inline index.js:12]: :100: nice addition, bonus for updating tests as well!

File: package.json [hunk 2]
```diff
@@ -23,19 +26,17 @@
   "license": "MIT",
   "main": "index.js",
   "maintainers": [
-    "Jack Boberg <jack.boberg@progress.com>"
+    "Jack Boberg <hello@jackboberg.info>",
+    "Jeremiah Harlan <jeremiah.harlan@gmail.com>"
```
Comment (jackboberg) [inline package.json:30]: 🥇

**Review by jackboberg — APPROVED** [review]
Thanks for this. I will need to change up the package name (and related links)..but I will deal with that in next PR.

**Pull request merged**